xdsl/transforms/canonicalization_patterns/csl_stencil.py

```diff
@@ -0,0 +1,33 @@
+from xdsl.dialects import tensor
+from xdsl.dialects.csl import csl_stencil
+from xdsl.ir import OpResult
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class RedundantIterArgInitialisation(RewritePattern):
+    """
+    Removes redundant allocations of empty tensors with no uses other than passed
+    as `iter_arg` to `csl_stencil.apply`. Prefer re-use where possible.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, op: csl_stencil.ApplyOp, rewriter: PatternRewriter
+    ) -> None:
```

Comment:
Would match all `csl_stencil.ApplyOp`s? I am guessing this doesn't really matter, since only the first one will get rewritten?

Reply:
It actually just recycles the arg being used.